spec/prawn/images_spec.rb

```diff
@@ -43,11 +43,46 @@
     expect(info.height).to eq(453)
   end
 
+  it 'does not close passed-in IO objects' do
+    file = File.open(filename, 'rb')
+    _info = pdf.image(file)
+
+    expect(file).to_not be_closed
+  end
+
   it 'accepts Pathname objects' do
     info = pdf.image(Pathname.new(filename))
     expect(info.height).to eq(453)
   end
 
+  it 'closes opened files again after getting Pathname objects' do
+    if RUBY_PLATFORM == 'java'
+      # JRuby doesn't support ObjectSpace
+      expect(RUBY_PLATFORM).to eq('java')
```

Comment:
There needs to be a spec here. Otherwise we can't be sure it works on Ruby.

Reply:
I'll try to come up with something... The spec that used `lsof` isn't really great either, that doesn't test anything for Windows platforms. Or would that work for you?

Reply:
Right now we don't run tests on Windows but we don't have any platform-specific code either. Everything is Ruby-only so in theory we shouldn't have any major issues with adding Windows. It at all possible I'd prefer it stayed that way.